source/magpie/dataframe.d

```diff
@@ -895,16 +902,30 @@ public:
     auto at(size_t i1, size_t i2) @property
     {
         assert(i2 < cols && i1 < rows, "Index out of bound");
-        static if(isHomogeneousType)
-            return data[i2][i1];
-        else
+
+        auto auxDispatch(alias Fn)(size_t indx)
```

Comment:
no need to declare this multiple times just use the one at the top. If these are on different types move this to you utils file and parameterise on the type.

The whole point was to reduce code remember ;)

Reply:
ditto throughout.

Reply:
I tried this but it leads to this error:
```
Error: function `magpie.dataframe.DataFrame!(int, 2).DataFrame.auxDispatch!(maxGapCalc).auxDispatch` cannot get frame pointer to magpie.dataframe.DataFrame!(int, 2).DataFrame.display.maxGapCalc!-1L.maxGapCalc
```

The function seems to accessible only within the function scope and couldn't be referenced outside hence I had to bring dispatcher into each function.
Is there a workaround for this?

Reply:
Try making auxDispatch `static`.

Reply:
making it static gives this:
```
Error: function `magpie.dataframe.DataFrame!(int, 2).DataFrame.display.maxGapCalc!-1L.maxGapCalc` is a nested function and cannot be accessed from magpie.dataframe.DataFrame!(int, 2).DataFrame.auxDispatch!(maxGapCalc).auxDispatch
```
telling nested function cannot be accessed outside.
I fiddled around this for a while however everything led to the same error

Reply:
Hmm, maybe try asking on the learn forums. There definitely _should_ be a way to do it.

Reply:
I was trying to make a sample example for dlang learn and this worked somehow
https://run.dlang.io/is/EvwVVq
So I was wondering what is the difference between this example and the one I was working with and I noticed if the nested function is a template function, it cannot send a pointer outside.

Reply:
I noticed it can if you instantiate it at the call site, i.e. this works (but is not what you want):
```diff
-void nested(int a)
+void nested()(int a)
...
-dispatcher!(nested)(5);
+dispatcher!(nested!())(5);
```

Reply:
```
struct S
{
    int data;
    int pos = 5;
private:  
    mixin template auxdispatch(alias F/*, alias indx*/)
    {
        auto auxdispatch(int x /*= indx*/)
        {
            return F(x);
        }
    }
 public:   
    void outer()
    {
        void nested()(int a)
        {
            data += a;
        }
        
        mixin auxdispatch!(nested/*,pos*/);
		auxdispatch(pos);
        import std.stdio;
        writeln(data);
    }
}

void main()
{
    S s;
    s.outer();
}
```

will do the trick, the commented out sections are what I would have liked to have done but a DMD bug prevents it from compiling.

Reply:
Thanks a lot for the above example @thewilsonator
Added mixin template in 403aa24